### ISSUE TITLE
[Discussion] New augmentations

### ISSUE THREAD

**Issue by NotNANtoN** [lucidrains/deep-daze#66]
Hey,

since we had the long discussion on the random cutout size sampling in #61 I played around a bit more with it. My issue with the current state of affairs is that for e.g. "A llama wearing a scarf and glasses, sitting in a cozy cafe." there will be many llamas appearing and also the general texture of the llama fur will be placed everywhere. That makes sense: we optimize the SIREN network to maximize the CLIP similarity at randomly sampled squares of sizes down to 10% of the original size. As it is not possible to train without these augmentations (it would only generate a weird-looking adversarial example), I thought of using other augmentations.

I tried adding Gaussian noise to the image - it is better than not adding any augmentations, but not good.

Then I tried sampling around a normal distribution with mean 0.6 and std of 0.2 instead of a uniform distribution between 0.1 and 1.0. The images look quite different - the background looks less interesting as the low-level textures get less weight. But I can't say they definitely look "better".

Lastly, I tried a simple change. Instead of averaging the loss over all random cutouts, I averaged the features of all cutouts to calculate a single loss. Here's where it gets interesting. The generated images look quite different now, often depicting some clear scenes of locations. What is strange, is there are some symbols that appear repeatedly over different prompts - a red and green blob right next to each other and also some kind of company logo appears quite regularly.

These are the results for the current version:
![depression_uniform](https://user-images.githubusercontent.com/19983153/110008592-2fd55b00-7d1c-11eb-9383-0ead88308bb0.jpeg)
![consciousness_uniform](https://user-images.githubusercontent.com/19983153/110008597-31068800-7d1c-11eb-8461-0398278af4fd.jpeg)
![schizophrenia_uniform](https://user-images.githubusercontent.com/19983153/110008600-31068800-7d1c-11eb-9cf1-58d7e41348d3.jpeg)
![lsd_uniform](https://user-images.githubusercontent.com/19983153/110008603-319f1e80-7d1c-11eb-9995-6ac09fd11089.jpeg)
![demon_uniform](https://user-images.githubusercontent.com/19983153/110008605-3237b500-7d1c-11eb-80c5-d65b8fb33acf.jpeg)
![lama_uniform](https://user-images.githubusercontent.com/19983153/110008609-3237b500-7d1c-11eb-9b06-f692b93283ec.jpeg)

And these are the results for averaged features:
![depression_averaged](https://user-images.githubusercontent.com/19983153/110008804-73c86000-7d1c-11eb-817c-3662708871eb.jpeg)
![consciousness_averaged](https://user-images.githubusercontent.com/19983153/110008808-7460f680-7d1c-11eb-8adf-ff7c65a9af23.jpeg)
![schizophrenia_averaged](https://user-images.githubusercontent.com/19983153/110008811-74f98d00-7d1c-11eb-8e9f-1f53b475b173.jpeg)
![lsd_averaged](https://user-images.githubusercontent.com/19983153/110008815-74f98d00-7d1c-11eb-8bc7-5294e9a1fb36.jpeg)
![demon_averaged](https://user-images.githubusercontent.com/19983153/110008818-75922380-7d1c-11eb-8823-0e59edb497ff.jpeg)
![lama_averaged](https://user-images.githubusercontent.com/19983153/110008820-75922380-7d1c-11eb-823b-a7a398c08a2f.jpeg)

**Comment by NotNANtoN** [comment]
@dginev @afiaka87, you might be interested in this!

**Comment by dginev** [comment]
Category | depression | consciousness | schizophrenia | lsd | demon | llama 
:--------|:-----------|:--------------|:--------------|:----|:------|:-----
Gaussian | ![depression_gauss_fixed](https://user-images.githubusercontent.com/19983153/110008960-a40ffe80-7d1c-11eb-8847-1aca1b58c850.jpg) | ![Consciousness_gauss_fixed](https://user-images.githubusercontent.com/19983153/110008965-a4a89500-7d1c-11eb-8c56-f70b097ee504.jpg) | ![Schizophrenia_gauss_fixed](https://user-images.githubusercontent.com/19983153/110008967-a5412b80-7d1c-11eb-80ab-826c022778bf.jpg) | ![lsd_gauss_fixed](https://user-images.githubusercontent.com/19983153/110008969-a5412b80-7d1c-11eb-9dd0-51f0f598e2e5.jpg) | ![demon_gauss_fixed](https://user-images.githubusercontent.com/19983153/110008970-a5d9c200-7d1c-11eb-9909-9ef8ad58f29d.jpg) | ![lama_gauss_fixed](https://user-images.githubusercontent.com/19983153/110008971-a5d9c200-7d1c-11eb-8bd1-8969fec94e40.jpg)
current version | ![depression_uniform](https://user-images.githubusercontent.com/19983153/110008592-2fd55b00-7d1c-11eb-9383-0ead88308bb0.jpeg) | ![consciousness_uniform](https://user-images.githubusercontent.com/19983153/110008597-31068800-7d1c-11eb-8461-0398278af4fd.jpeg) | ![schizophrenia_uniform](https://user-images.githubusercontent.com/19983153/110008600-31068800-7d1c-11eb-9cf1-58d7e41348d3.jpeg) | ![lsd_uniform](https://user-images.githubusercontent.com/19983153/110008603-319f1e80-7d1c-11eb-9995-6ac09fd11089.jpeg) | ![demon_uniform](https://user-images.githubusercontent.com/19983153/110008605-3237b500-7d1c-11eb-80c5-d65b8fb33acf.jpeg) | ![lama_uniform](https://user-images.githubusercontent.com/19983153/110008609-3237b500-7d1c-11eb-9b06-f692b93283ec.jpeg) 
averaged features | ![depression_averaged](https://user-images.githubusercontent.com/19983153/110008804-73c86000-7d1c-11eb-817c-3662708871eb.jpeg) | ![consciousness_averaged](https://user-images.githubusercontent.com/19983153/110008808-7460f680-7d1c-11eb-8adf-ff7c65a9af23.jpeg) | ![schizophrenia_averaged](https://user-images.githubusercontent.com/19983153/110008811-74f98d00-7d1c-11eb-8e9f-1f53b475b173.jpeg) | ![lsd_averaged](https://user-images.githubusercontent.com/19983153/110008815-74f98d00-7d1c-11eb-8bc7-5294e9a1fb36.jpeg) | ![demon_averaged](https://user-images.githubusercontent.com/19983153/110008818-75922380-7d1c-11eb-8823-0e59edb497ff.jpeg) | ![lama_averaged](https://user-images.githubusercontent.com/19983153/110008820-75922380-7d1c-11eb-823b-a7a398c08a2f.jpeg) 
Gaussian + feature averaging | ![depression_gauss_fixed_averaged](https://user-images.githubusercontent.com/19983153/110009100-d588ca00-7d1c-11eb-9364-72db15ce39c6.jpg) | ![consciousness_gauss_fixed_averaged](https://user-images.githubusercontent.com/19983153/110009102-d6216080-7d1c-11eb-9157-441cddb43c54.jpg) | ![schizophrenia_gauss_fixed_averaged](https://user-images.githubusercontent.com/19983153/110009104-d6216080-7d1c-11eb-9ff0-ff6defe78d85.jpg) | ![lsd_gauss_fixed_averaged](https://user-images.githubusercontent.com/19983153/110009108-d6b9f700-7d1c-11eb-8929-d5e99b0da191.jpg) | ![demon_gauss_fixed_averaged](https://user-images.githubusercontent.com/19983153/110009111-d6b9f700-7d1c-11eb-8f79-e8d8ca9b03e1.jpg) | ![lama_gauss_fixed_averaged](https://user-images.githubusercontent.com/19983153/110009114-d7528d80-7d1c-11eb-8798-084abecc85c6.jpg) 

Easy reply first - you can use a markdown table with the markdown images. Or just if you want to fine-tune widths, you can write an HTML table with `<img>` elements, both work on Github. Here's a quick scrape of your links and adding them into a markdown table.
